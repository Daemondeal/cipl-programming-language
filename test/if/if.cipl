if true: print("good")
if false: print("bad")

if true:
    print("block")

let a = false

if a = true:
    print(a)