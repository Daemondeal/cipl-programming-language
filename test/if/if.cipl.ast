(if true (then (expression (call print good)))
(if false (then (expression (call print bad)))
(if true (then (block
    (expression (call print block))
))
(let a false)
(if (assign a true) (then (block
    (expression (call print a))
))

