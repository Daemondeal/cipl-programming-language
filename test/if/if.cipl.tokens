Token          Lexeme         Line 
IF             if             1    
TRUE           true           1    
COLON          :              1    
IDENTIFIER     println        1    
LEFT_PAREN     (              1    
STRING         "good"         1     good
RIGHT_PAREN    )              1    
LINE_END       \n             2    
IF             if             2    
FALSE          false          2    
COLON          :              2    
IDENTIFIER     println        2    
LEFT_PAREN     (              2    
STRING         "bad"          2     bad
RIGHT_PAREN    )              2    
LINE_END       \n             3    
IF             if             4    
TRUE           true           4    
COLON          :              4    
LINE_END       \n             5    
INDENT                        5    
IDENTIFIER     println        5    
LEFT_PAREN     (              5    
STRING         "block"        5     block
RIGHT_PAREN    )              5    
LINE_END       \n             6    
DEDENT                        7    
LET            let            7    
IDENTIFIER     a              7    
EQUAL          =              7    
FALSE          false          7    
LINE_END       \n             8    
IF             if             9    
IDENTIFIER     a              9    
EQUAL          =              9    
TRUE           true           9    
COLON          :              9    
LINE_END       \n             10   
INDENT                        10   
IDENTIFIER     println        10   
LEFT_PAREN     (              10   
IDENTIFIER     a              10   
RIGHT_PAREN    )              10   
LINE_END       \n             10   
DEDENT                        10   
EOF                           10   
