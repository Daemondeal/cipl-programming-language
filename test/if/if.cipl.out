::stdout::
good
block
true

::stderr::
